wire/test/msgtest.go

```diff
@@ -12,29 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package wire
+package test
 
 import (
 	"io"
 	"testing"
 
-	"perun.network/go-perun/wire/perunio/test"
+	"perun.network/go-perun/wire"
 )
 
 type serializerMsg struct {
-	Msg Msg
+	Msg wire.Msg
 }
 
 func (msg *serializerMsg) Encode(writer io.Writer) error {
-	return Encode(msg.Msg, writer)
+	return wire.Encode(msg.Msg, writer)
 }
 
 func (msg *serializerMsg) Decode(reader io.Reader) (err error) {
-	msg.Msg, err = Decode(reader)
+	msg.Msg, err = wire.Decode(reader)
 	return err
 }
 
-// TestMsg performs generic tests on a wire.Msg object.
-func TestMsg(t *testing.T, msg Msg) {
-	test.GenericSerializerTest(t, &serializerMsg{msg})
+// MsgSerializerTest performs generic serializer tests on a wire.Msg object.
+func MsgSerializerTest(t *testing.T, msg wire.Msg) {
```

Comment:
```suggestion
func TestMsgSerialization(t *testing.T, msg wire.Msg) {
```

Reply:
My understanding is that, this is a helper function that runs serialization tests for any implementation of the interface `Msg`. In the same way as `GenericSerializerTest` or `GenericMarshalerTest`. Hence, I followed the same conventions for those two functions and

1. Named it this way.
2. Put it in a file name appended with test.
3. Marked it as helper.

Is this not the case ?

Reply:
**Regarding Helper**
[Helper doc](https://pkg.go.dev/testing#T.Helper): Helper marks the calling function as a test helper function. When printing file and line information, that function will be skipped. Helper may be called simultaneously from multiple goroutines.

I think this is not what we want here.

**Regarding the function name**
Before it was called `TestMsg`. The point is our naming scheme is not consistent anyway. And according to go guidelines function names should naturally read well as if you would read a sentence, which `MsgSerializerTest` does not, but `TestMsgSerialization ` does.

**Regarding the file name**
You didn't give a reason specifically why you appended "test" to the file name, so I can only restate that I think it's not needed as the package is already called "test".

Reply:
On marking this as helper, we do not want the test runner to report this line, rather the actual line where this is being called.
For instance, see the below example, where I intentionally failed the tests:

1. When we mark it as a helper function, the runner reports failure in the actual tests (proposalmsgs_test.go:170).

```
❯ got -run TestChannelProposalAccSerialization/ledger_channel
pkg/test: using rootSeed 1639049305959782908
--- FAIL: TestChannelProposalAccSerialization (0.00s)
    --- FAIL: TestChannelProposalAccSerialization/ledger_channel (0.00s)
        proposalmsgs_test.go:170:
                Error Trace:    serializertest.go:31
                                                        msgtest.go:40
                                                        proposalmsgs_test.go:170
                Error:          Not equal:
                                .........
                Test:           TestChannelProposalAccSerialization/ledger_channel
FAIL
exit status 1
FAIL    perun.network/go-perun/client   0.010s                                            /0.4s
```

2. When we don't mark it as helper, it reports the test is in `msgtest.go:40`, which is false, because the actual test case is elsewhere.
```
❯ got -run TestChannelProposalAccSerialization/ledger_channel
pkg/test: using rootSeed 1639049311486872753
--- FAIL: TestChannelProposalAccSerialization (0.00s)
    --- FAIL: TestChannelProposalAccSerialization/ledger_channel (0.00s)
        msgtest.go:40:
                Error Trace:    serializertest.go:31
                                                        msgtest.go:40
                                                        proposalmsgs_test.go:170
                Error:          Not equal:
                                .......
                Test:           TestChannelProposalAccSerialization/ledger_channel
FAIL
exit status 1
FAIL    perun.network/go-perun/client   0.010s                                            /0.4s
```

However, in both cases, the complete path will be shown in the Error Trace field.

Reply:
Regarding the name, if we being a function with `Test...` it means its a test function. But, in our case it is only a helper to run a set of generic tests. Also, the function would be read as `test.TestMsgSerialization`, which I think would cause a stutter.

Reply:
Regarding the filename, the package contains two kinds of things:
1. Things that can be used in tests. These are placed in  `randomizer.go` and `serializinglocalbus.go`.
2. Helper functions that run generic tests. These are placed in `bustest.go`, `marshalertest.go` and `serializertest.go`.

Hence, I think we should name it `msgtest.go`, as it falls in the second category.

Reply:
**Regarding Helper**
OK, your example convinced me. I thought it was then not printing the line where the error occurred. 👍 

**Regarding the function name**
OK.

**Regarding the file name**
OK.